actions/dependabotdotnet/src/dependabotdotnet/Program.cs

```diff
@@ -0,0 +1,224 @@
+﻿if (args is { Length: 0 } || args[0] is not string path)
+{
+    WriteLine("Must specify a repo root directory as input");
+    return 1;
+}
+
+string destinationFilePath = args is { Length: 2 } 
+    ? args[1] 
+    : $".github{Path.AltDirectorySeparatorChar}dependabot.yml";
+
+static void WriteLineToBufferAndOutput(StringBuilder buffer, string content)
+{
+    buffer.AppendLine(content);
+    WriteLine(content);
+}
+
+StringBuilder buffer = new();
+
+// yaml top-matter
+string topMatter = """
+    # generated by dependadotnet
+    # https://github.com/dotnet/core/tree/main/samples/dependadotnet
+    version: 2
+    updates:
+    """;
```

Comment:
May want to include the actions part too https://github.com/dotnet/core/pull/7845